### DISCUSSION TITLE
Keep cask up-to-date its auto-update or using upgrade --greedy?

### DISCUSSION THREAD

**Discussion by nesdroc** [Homebrew/discussions#2923] [Casks] [answered]
Is is better to keep a cask updated through itself (e.g. Signal with automatically checks for updates) or is it better to keep them up to date with brew.sh upgrade --greedy?

I'm thinking that perhaps a cask that keeps itself up to date would be able to at some point install things that brew.sh would later be unaware of when trying to use brew.sh to remove the cask at a later date. Whereas if using brew.sh to also keep these casks up to date it would know exactly what to remove again.

**Comment by SMillerDev** [comment, marked as answer]
Brew just runs the installer and uninstaller. Beyond that keeping track of files to Zap is up to contributors to notice.

If your brew is up to date it'll always know how to attempt an uninstall and the results won't differ if you use a different update mechanism.

The only benefit for greedy updates is that it also updates if you don't have the app open.

**Reply by nesdroc** [reply]
Thank you. This was very helpful. Was specifically the Zap I had noticed in the cask code. Also, wasn't sure how supporting files outside the app (e.g. Signal) itself would be treated in terms updating through the app itself vs. brew.sh upgrading.
